### PR TITLE
Matlab Memory Leak Fixes

### PR DESCRIPTION
**Pull request by varunagrawal**
@mattking-smith and I had a very productive session today identifying various issues in the Matlab wrapper. This PR addresses the memory leak Matt has been encountering in his code.

@mattking-smith can you pull these changes into GTSAM and verify this fix works?

### CODE REVIEW

**Comment by mattking-smith** [comment]
> @mattking-smith and I had a very productive session today identifying various issues in the Matlab wrapper. This PR addresses the memory leak Matt has been encountering in his code.
> 
> @mattking-smith can you pull these changes into GTSAM and verify this fix works?

I am working on this now

File: gtwrap/matlab_wrapper/wrapper.py [hunk 1]
```diff
@@ -1269,9 +1269,9 @@ def generate_collector_function(self, func_id):
                     Collector_{class_name}::iterator item;
                     item = collector_{class_name}.find(self);
                     if(item != collector_{class_name}.end()) {{
-                      delete self;
                       collector_{class_name}.erase(item);
                     }}
```
Comment (mattking-smith) [inline gtwrap/matlab_wrapper/wrapper.py:1273]: @varunagrawal Is there a reason we are keeping the collector in these lines of code? Or is this now obsolete code>

**Review by dellaert — APPROVED** [review]
Very cool!

**Review by mattking-smith — APPROVED** [review]
@varunagrawal @dellaert I have successfully tested and verified this MATLAB memory leak fix with both GTSAM and GPMP2! Well done @varunagrawal! 

I am able to release memory usage when calling GTSAM or GPMP2 code with `clear all`. 

Also, it appears that there may be increase performance. Seems to me that some tests are significantly faster, but I should investigate more thoroughly.  

I have one outstanding question. Is the following code necessary?
  ```
Collector_{class_name}::iterator item;
item = collector_{class_name}.find(self);
if(item != collector_{class_name}.end()) {{
collector_{class_name}.erase(item);
}}
```
Why is it necessary to have a collector of all the pointers if we let MATLAB actively delete the pointers? Even if I call `clear mex`, the pointers will already have been deleted, correct?

**Comment by mattking-smith** [comment]
I believe this pull request should resolve  issue: https://github.com/borglab/wrap/issues/48 when merged. After reviewing current MATLAB implementation of C++ classes with @varunagrawal, I feel we are appropriately resolving the outstanding issue.

**Comment by varunagrawal** [comment]
Yeah that collector stuff needs to be investigated for usefulness. I didn't want to change too much in this PR only to find out later that removing the collectors is causing issues elsewhere. That will be a subsequent, lower priority PR. :slightly_smiling_face: 

PS #48 is related to pointer handling and we just kind of segued into memory leaks in that issue. This PR should close #133

**Pull request merged**